File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int count = 0;

        for (List<String> item : items) {
            String type = item.get(0);
            String color = item.get(1);
            String name = item.get(2);

            if ((ruleKey.equals("type") && ruleValue.equals(type)) ||
                (ruleKey.equals("color") && ruleValue.equals(color)) ||
                (ruleKey.equals("name") && ruleValue.equals(name))) {
                count++;
            }
        }

        return count;
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
    let count = 0;
    for(let i=0;i<items.length;i++){
        if(ruleKey === "type" && items[i][0] === ruleValue){
            count++;
        }
        if(ruleKey === "color" && items[i][1] === ruleValue){
            count++;
        }
        if(ruleKey === "name" && items[i][2] === ruleValue){
            count++;
        }
    }
    return count;
};